cpp/src/parquet/encryption/encryption.cc

```diff
@@ -130,11 +123,11 @@ FileDecryptionProperties::Builder* FileDecryptionProperties::Builder::aad_prefix
 }
 
 ColumnDecryptionProperties::Builder* ColumnDecryptionProperties::Builder::key(
-    const std::string& key) {
+    encryption::SecureString key) {
   if (key.empty()) return this;
 
-  DCHECK(!key.empty());
-  key_ = key;
+  DCHECK(key_.empty());
```

Comment:
This looked suspicious, all other setter methods check if the member is unset. Here, the check is always true as it checks the input.

This is a breaking change if user code calls this setter twice with non-empty keys.

Reply:
I think you've corrected a bug; the intention with these setter methods seems to be one-time setting